// Copyright (c) 2021 Huawei Device Co., Ltd.
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
option java_package = "ohos.devtools.datasources.transport.grpc.service";
option optimize_for = LITE_RUNTIME;

// For Device Stat all below.
message AudioVolumeInfo {
    int32 stream_default = 1;
    int32 voice_call = 2;
    int32 music = 3;
    int32 stream_ring = 4;
    int32 media = 5;
    int32 voice_assistant = 6;
    int32 system = 7;
    int32 alarm = 8;
    int32 notification = 9;
    int32 bluetoolth_sco = 10;
    int32 enforced_audible = 11;
    int32 stream_dtmf = 12;
    int32 stream_tts = 13;
    int32 accessibility = 14;
    int32 recording = 15;
    int32 stream_all = 16;
}

// NODE Save the Device Stat
message DeviceStat {
    uint32 brightness_state = 1;
    int32 bt_state = 2;
    bool location_state = 3;
    int32 wifi_state = 4;
    AudioVolumeInfo volume_state = 5;
}

// NODE Save the captured data
message HisyseventLine {
    string raw_content = 1;
    uint64 id = 2;
}

message HisyseventInfo {
    repeated HisyseventLine info = 1;
    DeviceStat device_state = 2;
}